### PR TITLE
dev_guide: merge doc related to sanity tests

### PR DESCRIPTION
**Pull request by pilou-**
##### SUMMARY
`dev_guide`: merge `testing/sanity/index.rst` and `testing_sanity.rst`

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
dev_guide

##### ANSIBLE VERSION
```
ansible-playbook 2.5.0 (devel 588bddbc35) last updated 2017/09/08 17:06:00 (GMT +200)
```

### CODE REVIEW

**Review by dharmabumstead — APPROVED** [review]
Thanks @pilou-

**Comment by pilou-** [comment]
Could this one be merged ?

**Comment by bcoca** [comment]
cc @mattclay

**Comment by mattclay** [comment]
Instead of continuing to maintain this page, we should migrate any relevant content to https://docs.ansible.com/ansible/devel/dev_guide/testing/sanity/ and associated pages.

The script https://github.com/ansible/ansible/blob/devel/docs/bin/testing_formatter.sh is used to generate that page since it has a dynamic index based on available sanity tests.

**Comment by pilou-** [comment]
@mattclay thanks for pointing that! I merged `testing/sanity/index.rst` and `testing_sanity.rst`.

File: docs/bin/testing_formatter.sh [hunk 1]
```diff
@@ -1,12 +1,47 @@
 #!/bin/bash -eu
 
-cat <<- EOF > ../docsite/rst/dev_guide/testing/sanity/index.rst
+cat <<- 'EOF' > ../docsite/rst/dev_guide/testing_sanity.rst
```
Comment (mattclay) [inline docs/bin/testing_formatter.sh:3]: I think it makes more sense to move the testing_sanity contents to the sanity index.

@dharmabumstead Thoughts?

Comment (pilou-) [reply docs/bin/testing_formatter.sh:3]: I might have missed something but it seems `sanity/index.rst` wasn't referenced anywhere.

Comment (mattclay) [reply docs/bin/testing_formatter.sh:3]: I think it should be referenced instead of the old testing_sanity page.

Comment (pilou-) [reply docs/bin/testing_formatter.sh:3]: Where should be put the reference ?

Note that `dev_guide/testing/compile/index.rst` isn't referenced too and refers to `testing_compile` (this one is referenced).
```
Compile Tests
=============

See :doc:`../../testing_compile` for more information.
```

Comment (dharmabumstead) [reply docs/bin/testing_formatter.sh:3]: @mattclay that works for me.

**Comment by dharmabumstead** [comment]
@mattclay is this good to go?

**Comment by mattclay** [comment]
@dharmabumstead The sanity docs still need to be moved to back to `sanity/index.rst`.

**Comment by dharmabumstead** [comment]
Closing this one. Feel free to rebase and reopen.

**Pull request closed without merging**